bgpd/bgp_route.c

```diff
@@ -2044,10 +2053,25 @@ bool subgroup_announce_check(struct bgp_node *rn, struct bgp_path_info *pi,
 	 * the same interface.
 	 */
 	if (p->family == AF_INET6 || peer_cap_enhe(peer, afi, safi)) {
-		if (IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_global))
+		if (IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_global)) {
 			subgroup_announce_reset_nhop(AF_INET6, attr);
+				nh_reset = true;
+			}
 	}
 
+	/*
+	 * When the next hop is set to ourselves, if all multipaths have
+	 * link-bandwidth announce the cumulative bandwidth as that makes
+	 * the most sense. However, don't modify if the link-bandwidth has
+	 * been explicitly set by user policy.
+	 */
+	if (nh_reset &&
+	    bgp_path_info_mpath_chkwtd(bgp, pi) &&
+	    (cum_bw = bgp_path_info_mpath_cumbw(pi)) != 0 &&
+	    !CHECK_FLAG(attr->rmap_change_flags, BATTR_RMAP_LINK_BW_SET))
+		attr->ecommunity = ecommunity_replace_linkbw(
```

Comment:
nh_reset is for `Additionally, if a
   route is received with link bandwidth extended community and the BGP
   speaker sets itself as next-hop while announcing that route to other
   peers, the link bandwidth extended community should be removed.`, right? shouldn't extcommunity be removed in this case?

Reply:
No, this is for the case where we announce the cumulative link-bandwidth of all the downstream multipaths, instead of just the best-path's bandwidth. This is the use case in https://tools.ietf.org/html/draft-mohanty-bess-ebgp-dmz